esp/smc/SMCLib/TpContainer.cpp

```diff
@@ -0,0 +1,781 @@
+    const char* Status, const char* Directory, IArrayOf<IEspTpMachine>& MachineList, set<string>* pMachineNames/*=NULL*/)
+{
+    IWARNLOG("UNIMPLEMENTED: CONTAINERIZED(CTpWrapper::getMachineList)");
+}
+
+const char* CTpWrapper::getNodeNameTag(const char* MachineType)
+{
+    if (strcmp(MachineType,"Computer")==0)
+        return "@name";
+    else
+        return "@computer";
+}
+
+void CTpWrapper::getDropZoneMachineList(double clientVersion, bool ECLWatchVisibleOnly, IArrayOf<IEspTpMachine> &MachineList)
+{
+    try
+    {
+        IArrayOf<IConstTpDropZone> list;
+        getTpDropZones(clientVersion, nullptr, ECLWatchVisibleOnly, list);
+        ForEachItemIn(i, list)
+        {
+            IConstTpDropZone& dropZone = list.item(i);
+
+            IArrayOf<IConstTpMachine>& tpMachines = dropZone.getTpMachines();
+            ForEachItemIn(ii, tpMachines)
+            {
+                IConstTpMachine& tpMachine = tpMachines.item(ii);
+                Owned<IEspTpMachine> machine = createTpMachine();
+                machine->copy(tpMachine);
+
+                MachineList.append(*machine.getLink());
+            }
+        }
+    }
+    catch(IException* e)
+    {
+        EXCLOG(e);
+        e->Release();
+    }
+    catch(...)
+    {
+        IWARNLOG("Unknown Exception caught within CTpWrapper::getDropZoneMachineList");
+    }
+}
+
+//For a given dropzone or every dropzones (check ECLWatchVisible if needed), read: "@name",
+// "@description", "@build", "@directory", "@ECLWatchVisible" into an IEspTpDropZone object.
+//For each ServerList, read "@name" and "@server" (hostname or IP) into an IEspTpMachine object.
+//Add the IEspTpMachine object into the IEspTpDropZone.
+
+void CTpWrapper::getTpDropZones(double clientVersion, const char* name, bool ECLWatchVisibleOnly, IArrayOf<IConstTpDropZone>& list)
+{
+    Owned<IPropertyTreeIterator> planes = getDropZonePlanesIterator(name);
+    ForEach(*planes)
+    {
+        IPropertyTree & plane = planes->query();
+        const char * dropzonename = plane.queryProp("@name");
+        const char * path = plane.queryProp("@prefix");
+        Owned<IEspTpDropZone> dropZone = createTpDropZone();
+        dropZone->setName(dropzonename);
+        dropZone->setDescription("");
+        dropZone->setPath(path);
+        dropZone->setBuild("");
+        dropZone->setECLWatchVisible(true);
+        IArrayOf<IEspTpMachine> tpMachines;
+        gatherDropZoneMachines(tpMachines, plane);
+        dropZone->setTpMachines(tpMachines);
+        list.append(*dropZone.getClear());
+    }
+}
+
+void CTpWrapper::getTpSparkThors(double clientVersion, const char* name, IArrayOf<IConstTpSparkThor>& list)
+{
+    UNIMPLEMENTED_X("CONTAINERIZED(CTpWrapper::getTpSparkThors)");
+}
+
+void CTpWrapper::appendTpMachine(double clientVersion, IConstEnvironment* constEnv, IConstInstanceInfo& instanceInfo, IArrayOf<IConstTpMachine>& machines)
```

Comment:
Should the appendTpMachine() (and other 'common code') be moved to TpCommon.cpp?

Reply:
It should probably be removed from this file - it is only called from the CONTAINERIZED code.  I didn't go through this file and remove all code that is no longer required.